### DISCUSSION TITLE
Problem when trying to validate with Vuelidade

### DISCUSSION THREAD

**Discussion by douglas-88** [beholdr/maska#140] [Q&A]
Hello,

I have a form that I use vuelidate to validate (https://vuelidate-next.netlify.app/) .

And I can also make the mask work in the fields using the plugin (https://beholdr.github.io/maska/#/?id=live-demo) .

I'm on vue 3 .

The problem is that, for example, there is a field called "cnpj" with 14 digits, and I can never validate it, because the mask adds characters, and then it becomes 18 digits...

Is there a way to pass the raw value to vuelidate to validate correctly?

![Captura de tela de 2023-03-01 14-08-10](https://user-images.githubusercontent.com/10869960/222211539-e7113483-15f3-4a41-bfb1-12d879866754.png)

Esse é meu código:

`  <div class="row">
      <div class="col-4">
        <label for="cnpj" class="form-label">CNPJ:</label>
        <input v-maska data-maska="##.###.###/####-##" type="text" id="cnpj" name="cnpj" class="form-control"
               :class="this.v$.cnpj.$invalid ? 'is-invalid' : 'is-valid'" v-model="cnpj">
        <div :class="this.v$.cnpj.$invalid ? 'invalid-feedback' : 'valid-feedback'" v-for="erros in this.v$.cnpj.$errors">
          {{erros.$message}}
        </div>
        <br>
      </div>
`

Forgive my writing, as I'm using Google Translate.

**Comment by beholdr** [comment]
You could use [regex validator](https://vuelidate-next.netlify.app/custom_validators.html#regex-based-validator) to validate masked value?

**Reply by douglas-88** [reply]
But that way I will always have to use regex ?

I have other fields with the same issue as well.

**Reply by beholdr** [reply]
I think so, if vuelidate validates input value which is masked.
Or you could get unmasked value and validate it manually somehow.

**Reply by douglas-88** [reply]
How do I get the input value without the mask?

**Reply by beholdr** [reply]
Via binding to value:

``` html
<script setup>
import { reactive, ref } from "vue"
import { vMaska } from "maska"

const bindedObject = reactive({})
</script>

<template>
  <input v-maska="bindedObject">

  Unmasked value: {{ bindedObject.unmasked }}
</template>
```

or via `@maska` event
